session/session_test.go

```diff
@@ -70,6 +122,71 @@ func (s *SessionTestSuite) TestHas() {
 	s.False(s.session.Has("nilKey"))
 }
 
+func (s *SessionTestSuite) TestInvalidate() {
+	oldID := s.session.GetID()
+	s.session.Put("foo", "bar")
+	all := s.session.All()
+	s.Equal("bar", all["foo"])
+
+	s.session.Flash("name", "Krishan")
+	s.True(s.session.Has("name"))
+
+	s.driver.On("Destroy", oldID).Return(nil).Once()
+	s.Nil(s.session.Invalidate())
+
+	s.False(s.session.Has("name"))
+	s.NotEqual(oldID, s.session.GetID())
+	s.Equal(map[string]any{}, s.session.All())
+
+	oldID = s.session.GetID()
+	s.driver.On("Destroy", oldID).Return(errors.New("error")).Once()
+	s.Equal(errors.New("error"), s.session.Invalidate())
+	s.Equal(oldID, s.session.GetID())
+}
+
+func (s *SessionTestSuite) TestMigrate() {
+	oldID := s.session.GetID()
+	s.Nil(s.session.Migrate())
+	s.NotEqual(oldID, s.session.GetID())
+
+	oldID = s.session.GetID()
+	s.driver.On("Destroy", oldID).Return(nil).Once()
+	s.Nil(s.session.Migrate(true))
+	s.NotEqual(oldID, s.session.GetID())
+}
+
+func (s *SessionTestSuite) TestMissing() {
+	s.session.Put("foo", "bar")
+	s.False(s.session.Missing("foo"))
+	s.session.Put("baz", nil)
+	s.False(s.session.Has("baz"))
+	s.False(s.session.Missing("baz"))
```

Comment:
Ditto

Reply:
It is not repeating. The value of the key `baz` is `nil`, so `Has` should return `True`, and `Missing` is just the negation of `Exists`, so it should return `False`.